### DISCUSSION TITLE
volume control is not working on samsung smart monitor M8

### DISCUSSION THREAD

**Discussion by Phryxos** [MonitorControl/MonitorControl#1077] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Volume & contrast controls are not working on my display : Controls of Monitor Control look like they're acting perfectly fine, but do not actually change at all the volume nor the contrast. 

My display (smart Monitor M8) is  connected on USB-C (thunderbolt) port .
I've only one display connected, with speakers, mic & camera embedded in display.

### Expected behavior

just expect to manage volume with same effect as shown in Monitor Control dashboard

### Environment Information (please complete the following information)

```markdown
- macOS version:12.4 Monterey
- Mac model:MAc Studio M1 Max (2022)
- MonitorControl version: 4.1
- Monitor(s):Samsung  Smart Monitor M8 
- Monitor Cable(s)/Connection(s):usb C (thunderbolt/Display Port)
- Apple Silicon/M1 (yes or no):yes
```

**Comment by waydabber** [comment]
Hi there,

- Does brightness control work or only the bottom 50% of the scale does anything (if so then that is only software adjustment, not DDC control).
- Can you control the display using m1ddc or the [latest beta of BetterDummy](https://github.com/waydabber/BetterDummy/releases/tag/v1.2.3-beta)? If not then there might be some kind of display. incompatibility.

**Comment by Phryxos** [comment]
Hi,
thanks a lot for this quick feedback.
-unfortunately, brightness control works only for the bottom 50%
-just tried betterdummy latest beta : same issue , no control of the volume; only bottom 50% for brightness + screen resolution do work well.
I'll contact Samsung too, to know if there is a way to activate DDC on this display, if this is the issue. it's curious for a display that is really targetting Macbook & Mac studio users, including for Camera & Audio. they should keep a way to control it, but ...

**Comment by waydabber** [comment]
You might want to check for a DDC/CI toggle in the monitor OSD (but I don't think Samsungs have one). I don't have any other idea. You can try [m1ddc](https://github.com/waydabber/m1ddc) and see if it gives an actual error or not. If you have a Windows PC around, you can also try some apps like Monitorian and see if that works.

**Comment by stewartbryson** [comment]
Any update @Phryxos? I'm having the same issue, and I'm trying to decide whether to return the monitor or not.

**Comment by kevinwinsper** [comment]
Same issue on the predecessor of the m8, the samsung m7 monitor. I'm connecting via USB-C (DisplayPort).

**Comment by Phryxos** [comment]
> Any update @Phryxos? I'm having the same issue, and I'm trying to decide whether to return the monitor or not.

Hi,
I've tried several tools, as various parameters on the display... still doesn't work. Looks silly, but it actually seems that Samsung is blocking the DDC/CI protocol on smart monitors, with no way to activate it... I've contacted Samsung yesterday to know about possible software update on their side.  Expecting their reply ..

**Comment by waydabber** [comment]
Let us know if there is any response from them! I'll convert this one into a discussion as this comes up from time-to-time and  might be a good place to collect info about these displays.

**Comment by Phryxos** [comment]
Hi, 
Just received answer from Samsung. to make it short : "Samsung monitors are indeed managing CEC/DDC signals, so issue cannot come from SAMSUNG, but it's due to APPLE or the APPS you're running".  No will to analyse the topic, it's very disappointing. 

I've also tested with LUNAR , for which activation of SMART controls on Samsung monitors are known to block DDC signals, such as Input signal plus ; Magic bright ; eye saver mode ; eco saving plus ; smart eco saving ; game mode ; PIP/PBP mode ; dynamic brightness...

My feeling is that actually, the smart controls of the monitor are set with the highest priority/highest privileges, so they always take control over DDC signals from a MAC. And I did not find any way in Samsung MENU to completely disable all the smart controls ; well, it still was not working after deactivating those that are accessible in the MENU...

**Reply by waydabber** [reply]
well in this regard Lunar and MonitorControl is the same. If the display ignores or overrides DDC commands, things will not work. :(

**Comment by Phryxos** [comment]
that's what I fear indeed. We would need a Samsung specialist to find a way to disable these controls or reduce their level of priority ... 

any idea from any user of Samsung smart monitors would  be most welcome ! :-)

**Reply by MacsR4me** [reply]
Had the same issue with Samsung M8 and Magic Keyboard sound keys and https://staticz.com/soundcontrol/ trial version app worked for me. Haven't got full version.

**Reply by Phryxos** [reply]
indeed soundcontrol works for me too. Good news, it means there is a way to override samsung priorities or preferences.. if we can find a way to get this also in BetterDisplay to completely manage samsung Display,  it would be absolutely perfect !

**Reply by MacsR4me** [reply]
Agree!!

**Reply by instanceofme** [reply]
IIUC Sound Control for that case works by installing an audio driver and controlling the volume's software level, see https://staticz.zendesk.com/hc/en-us/articles/115002461132-Keyboard-volume-controls-don-t-work-with-my-device-Can-this-be-fixed-

Ditto for [eqMac](https://eqmac.app/), you can control the sound but it's software only.

It works but notably implies that
- you need to set the monitor's sound to the maximum volume (or the one you want to be 100%, and won't be able to go past it)
- using the monitor's "smart" features / changing the volume with the remote will mess with this
- unfortunately it's not an actual override of the monitor's priorities/preferences, just a software workaround

**Comment by ioswmtdeveloper** [comment]
https://github.com/Toxblh/samsung-tv-control

Guys this works perfectly for samsung smart monitor. I tested in samsung M5 monitor for sound control only. We need to develop electron app and wrap this nodejs framework inside ui.

**Comment by teekamsuthar** [comment]
Do we have any update on this issue here?

**Comment by Phryxos** [comment]
Hi,
I don't know if there are a latest version of Betterdummy that fixes the issue, butI'm still using sound control which works well.

**Comment by pankajet** [comment]
M8 (M80D) volume works for windows 11.

**Reply by waydabber** [reply]
Windows does not use DDC but simply uses a software mixer.

**Reply by pankajet** [reply]
ok. Is there fix for M8 yet ?

**Reply by waydabber** [reply]
If the M8 does not support DDC volume control, then there is no fix, only Samsung could provide an updated firmware.

**Comment by pankajet** [comment]
Is there further fix to enable DDC for M8 for Mac ?
